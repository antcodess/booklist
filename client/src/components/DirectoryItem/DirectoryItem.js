import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cart.actions';

import CustomButton from '../CustomButton/CustomButton';

import './DirectoryItem.scss';

const DirectoryItem = ({ item, history, addItem }) => {
  const { title, imageUrl, price } = item;
  return (
    <div className='directory-item'>
      <div className='directory-item-image' onClick={() => history.push(`/${item.routeName}`)}>
        <img className='image' src={ imageUrl } alt='products' />
      </div>
      <div className='directory-meta' onClick={() => history.push(`/${item.routeName}`)}>
        <h4>{ title }</h4>
        <span>${ price }</span>
      </div>
      <CustomButton onClick={() => addItem(item)}>Add To Cart</CustomButton>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(null, mapDispatchToProps)(withRouter(DirectoryItem));