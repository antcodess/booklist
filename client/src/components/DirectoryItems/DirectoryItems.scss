.directory-items-container {
  margin-bottom: 60px; 

  .directory-items-title {
    display: flex;

    .directory-items-title-link {
      align-items: center;
      font-family: Helvetica, Arial, "Lucida Grande", sans-serif;
      font-weight: bolder;
      font-size: 0.875rem;
      margin-right: 5px;

      &:hover {
        color: #0080DE;
      }
  
    }
    h3 {
      margin-bottom: 30px;
      cursor: pointer;

      &:hover {
        opacity: 0.7;
      }
    }
  }

  .directory-items {
    vertical-align: baseline;

    .slick-prev:before, .slick-next:before {
      // color: black;
      font-size:40px;
      opacity: 1;
    }

    .slick-slide {
      padding: 0 5px;
    }

    .slick-dots li button:before {
      font-size: 13px;
      line-height: 40px;
    }

    .slick-dots li.slick-active button:before {
      opacity: 1;
    }

    .slick-prev {
      left: -42px;
    }
  }
}

@media only screen and (max-width: 1500px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 125px;
      }
    }
  } 
}

@media only screen and (max-width: 1100px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 95px;
      }
    }
  }
}

@media only screen and (max-width: 600px) {
  .directory-items-container {
    h3 {
      justify-content: center;
    }
    .directory-items-title {
      justify-content: center;
    }
    .directory-items {
      .slick-prev, .slick-next{
        top: 155px;
      }
    }
  } 
}

@media only screen and (max-width: 500px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 130px;
      }
    }
  }
}

@media only screen and (max-width: 400px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 83px;
      }
    }
  } 
}