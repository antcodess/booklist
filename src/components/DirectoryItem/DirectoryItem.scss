.directory-item {
  position: relative;
  cursor: pointer;

  .directory-item-image {

    .image {
      border-radius: 10px;
      max-width: 100%;
      height: auto;
    }
  }

  h4 {
    margin-top: 10px;
    margin-bottom: 2px;
  }
  
  .custom-button {
    display: none;
    width: 90%;
    top: 58%;
    right: 5%;
    position: absolute;
  }

  &:hover {
    .custom-button {
      display: flex;
    }
    
    .image {
      opacity: 0.8;
    }
  }
}