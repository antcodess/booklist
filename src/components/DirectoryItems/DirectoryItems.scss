.directory-items-container {
  margin-bottom: 60px; 

  h3 {
    display: flex;
    margin-bottom: 30px;
  }

  .directory-items {
    // width: 100%;
    // display: flex;
    // justify-content: space-between;
    // flex-wrap: wrap;
    // .slick-prev {
    //   left: -32px;
    // }
    
    .slick-prev:before, .slick-next:before {
      color: black;
      // font-size: 28px;
      opacity: 1;
    }

    .slick-slide {
      padding: 0 5px;
    }

    // .slick-prev, .slick-next {
    //   top: 95px;
    // }

    .slick-dots li button:before {
      font-size: 13px;
      line-height: 40px;
    }

    .slick-dots li.slick-active button:before {
      opacity: 1;
    }
  }
}