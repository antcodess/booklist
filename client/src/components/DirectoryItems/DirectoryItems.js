import React from 'react';
import Slider from 'react-slick';
import { withRouter } from 'react-router-dom';

import DirectoryItem from '../DirectoryItem/DirectoryItem';

import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import './DirectoryItems.scss';


const DirectoryItems = ({ title, items, history, match }) => {
  
  const settings = {
    dots: true,
    infinite: items.length > 3 ? true : false,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 4,
    swipeToSlide: true,
    arrows: false,
    responsive: [
      {
        breakpoint: 940,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
        }
      },
      {
        breakpoint: 700,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
        }
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1
        }
      }
    ]
  }; 

  const categoryURL = encodeURIComponent(title.replace(/[\s]/g, '-').toLowerCase());

  return (
    <div className='directory-items-container'>
      <div className='directory-items-title'>
        <h3 onClick={() => history.push(`/categories/${categoryURL}`)}>
          { title }
        </h3>
      </div>
      <div className='directory-items'>
        <Slider { ...settings }>
          {items
            .filter((item, index) => index < 6)
            .map((item) => (
              <DirectoryItem key={item.id} item={item} />
            ))
          }
        </Slider>
      </div>
    </div>
  );
};

export default withRouter(DirectoryItems);