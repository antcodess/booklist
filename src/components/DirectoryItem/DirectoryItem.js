import React from 'react';


import './DirectoryItem.scss';

const DirectoryItem = ({ photo, name, price }) => {
  return (
    <div className='directory-item'>
      <div className='directory-item-image'>
        <img src={ photo } alt='products' />
      </div>
      <h4>{ name }</h4>
      <span>{ price }</span>
    </div>
  );
};

export default DirectoryItem;