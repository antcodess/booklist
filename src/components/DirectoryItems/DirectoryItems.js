import React from 'react';
import Slider from 'react-slick';
import { Link } from 'react-router-dom';
import DirectoryItem from '../DirectoryItem/DirectoryItem';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import './DirectoryItems.scss';


const DirectoryItems = ({ title, items }) => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 4,
    swipeToSlide: true,
    arrows: false,
    responsive: [
      {
        breakpoint: 900,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
        }
      },
      {
        breakpoint: 700,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
        }
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1
        }
      }
    ]
  };

  const categoryURL = title.replace(/[\s]/g, '-').toLowerCase();

  return (
    <div className='directory-items-container'>
      <div className='directory-items-title'>
        <h3>{ title }</h3>
        <Link className='directory-items-title-link' to={`categories/${categoryURL}`}>View All</Link>
      </div>
      <div className='directory-items'>
        <Slider { ...settings }>
          {items
            .filter((item, index) => index < 6)
            .map((item) => (
              <DirectoryItem key={item.id} item={item} />
            ))
          }
        </Slider>
      </div>
    </div>
  );
};

export default DirectoryItems;