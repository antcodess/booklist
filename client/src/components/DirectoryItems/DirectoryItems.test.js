import React from 'react';
import { shallow } from 'enzyme';
import { DirectoryItems } from './DirectoryItems';
import DirectoryItem from '../DirectoryItem/DirectoryItem';

describe('DirectoryItems Component', () => {
  let wrapper;
  let mockItems;
  let mockTitle;
  let mockHistory;
  let dummyFunction = jest.fn()
  
  beforeEach(() => {

    mockTitle = 'Random Title';
    
    mockItems = [
      {
        id: 1,
        title: 'Item 1', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-1'
      },
      {
        id: 2,
        title: 'Item 2', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-2'
      },
      {
        id: 3,
        title: 'Item 3', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-3'
      },
      {
        id: 4,
        title: 'Item 1', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-1'
      },
      {
        id: 5,
        title: 'Item 2', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-2'
      },
      {
        id: 6,
        title: 'Item 3', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-3'
      },
      {
        id: 7,
        title: 'Item 3', 
        imageUrl: 'example.com', 
        quantity: 1, 
        price: 20, 
        routeName: '/products/item-3'
      }
    ];
    
    mockHistory = { push: jest.fn() };

    const mockProps = {
      title: mockTitle, 
      items: mockItems, 
      history: mockHistory
    };

    wrapper = shallow(<DirectoryItems { ...mockProps } />);
  });

  it('should render DirectoryItems Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for h3', () => {
    wrapper.find('h3').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should render Directory Item if component has items', () => {
    expect(wrapper.exists(DirectoryItem)).toBe(true);
  });

  it('should render only 6 or less Directory Item components', () => {
    expect(wrapper.find(DirectoryItem).length).toEqual(6);
  });
  
  it('should not render a Directory Item if items are empty', () => {
    const mockProps = {
      title: mockTitle, 
      items: [], 
      history: mockHistory
    };
    wrapper = shallow(<DirectoryItems { ...mockProps } />)
    expect(wrapper.exists(DirectoryItem)).toBe(false);
    expect(wrapper.find('.directory-items').children().length).toEqual(1);
  });
});
