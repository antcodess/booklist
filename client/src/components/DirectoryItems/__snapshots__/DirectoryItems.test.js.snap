// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectoryItems Component should render DirectoryItems Component 1`] = `ShallowWrapper {}`;
